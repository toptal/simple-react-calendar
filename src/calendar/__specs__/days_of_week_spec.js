import React from 'react'

import DaysOfWeek from '../days_of_week'

import ReactTestRenderer from 'react-test-renderer'
import {shallow} from 'enzyme'

describe('DaysOfWeek', () => {
  let wrapper, tree

  describe('#_getDaysOfWeek', () => {
    it('return days of week', () => {
      wrapper = shallow(<DaysOfWeek weekStartsOn={2} />)

      const daysOfWeek = wrapper.instance()._getDaysOfWeek(3)

      expect(daysOfWeek).toEqual(['Wed', 'Thu', 'Fri', 'Sat', 'Sun', 'Mon', 'Tue'])
    })
  })

  describe('#render', () => {
    it('renders <DaysOfWeek />', () => {
      tree = ReactTestRenderer.create(<DaysOfWeek weekStartsOn={2} blockClassName="test-class" />).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
