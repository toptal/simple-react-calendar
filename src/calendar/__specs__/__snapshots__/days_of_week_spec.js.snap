// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DaysOfWeek #render renders <DaysOfWeek /> 1`] = `
<div
  className="test-class-days_of_week"
>
  <div
    className="test-class-days_of_week_day"
  >
    Tue
  </div>
  <div
    className="test-class-days_of_week_day"
  >
    Wed
  </div>
  <div
    className="test-class-days_of_week_day"
  >
    Thu
  </div>
  <div
    className="test-class-days_of_week_day"
  >
    Fri
  </div>
  <div
    className="test-class-days_of_week_day"
  >
    Sat
  </div>
  <div
    className="test-class-days_of_week_day is-weekend"
  >
    Sun
  </div>
  <div
    className="test-class-days_of_week_day is-weekend"
  >
    Mon
  </div>
</div>
`
